### PR TITLE
Add stretch (etc) support

### PR DESCRIPTION
**Pull request by dracos**
This updates carton/cpanm to deal with e.g. `.` not being in `@INC` in Perl 5.24, so that modules will still install okay. It removes modules from the snapshot that were only there for squeeze support (dropped a long time ago), fixes some stretch installation issues, and tests perl 5.24 on Travis (which is still running precise).

### CODE REVIEW

File: cpanfile [hunk 1]
```diff
@@ -11,6 +11,7 @@ requires 'ExtUtils::MakeMaker', '6.72'; # [1]
 requires 'Guard', '1.023';
 requires 'PadWalker', '2.2';
 requires 'aliased', '0.34';
+requires 'Net::SSLeay', '1.81';
```
Comment (davea) [inline cpanfile]: Woo, this seems to install cleanly on macOS! Now seems a good opportunity to upgrade `Net::Twitter::Lite::WithAPIv1_1` to 0.12008, which no longer depends on `Crypt::SSLeay` (the final macOS pain point) 😄

Comment (dracos) [inline cpanfile]: Your macOS, it's always installed fine on mine :P Sounds good, will stick that in.

Comment (dracos) [inline cpanfile]: I've added a commit to upgrade that and SOAP::Lite, which then drops Crypt::SSLeay entirely. Not sure the SOAP::Lite using things are well tested (Zurich geocoder, Angus/EastHants report submission) so might need careful deploy though nothing major in changelog I can see.

**Comment by dracos** [comment]
@davea Do f89681f and 01b426f look okay to you?

**Comment by davea** [comment]
Yep, they look fine to me! 👍

**Pull request merged**